numpy/__init__.pyi

```diff
@@ -386,133 +388,172 @@ class generic(_ArrayOrScalarCommon):
     @property
     def base(self) -> None: ...
 
-class _real_generic(generic):  # type: ignore
+class number(generic):  # type: ignore
     @property
     def real(self: _ArraySelf) -> _ArraySelf: ...
     @property
     def imag(self: _ArraySelf) -> _ArraySelf: ...
 
-class number(generic): ...  # type: ignore
-
-class bool_(_real_generic):
+class bool_(generic):
     def __init__(self, __value: object = ...) -> None: ...
+    @property
+    def real(self: _ArraySelf) -> _ArraySelf: ...
+    @property
+    def imag(self: _ArraySelf) -> _ArraySelf: ...
 
 class object_(generic):
     def __init__(self, __value: object = ...) -> None: ...
+    @property
+    def real(self: _ArraySelf) -> _ArraySelf: ...
+    @property
+    def imag(self: _ArraySelf) -> _ArraySelf: ...
 
-class datetime64:
+class datetime64(generic):
     @overload
     def __init__(
         self,
-        __value: Union[None, datetime64, str, dt.datetime] = ...,
-        __format: str = ...
+        __value: Union[None, datetime64, str, bytes, dt.datetime] = ...,
+        __format: Union[str, bytes] = ...,
     ) -> None: ...
     @overload
-    def __init__(self, __value: int, __format: str) -> None: ...
+    def __init__(self, __value: int, __format: Union[str, bytes]) -> None: ...
     def __add__(self, other: Union[timedelta64, int]) -> datetime64: ...
     def __sub__(self, other: Union[timedelta64, datetime64, int]) -> timedelta64: ...
+    def __rsub__(self, other: Union[datetime64, int]) -> timedelta64: ...
 
-class integer(number, _real_generic): ...  # type: ignore
+class integer(number): ...  # type: ignore
 class signedinteger(integer): ...  # type: ignore
 
 class int8(signedinteger):
-    def __init__(self, __value: SupportsInt = ...) -> None: ...
+    def __init__(
+        self, __value: Union[SupportsInt, SupportsIndex, str, bytes] = ...
+    ) -> None: ...
 
 class int16(signedinteger):
-    def __init__(self, __value: SupportsInt = ...) -> None: ...
+    def __init__(
+        self, __value: Union[SupportsInt, SupportsIndex, str, bytes] = ...
+    ) -> None: ...
 
 class int32(signedinteger):
-    def __init__(self, __value: SupportsInt = ...) -> None: ...
+    def __init__(
+        self, __value: Union[SupportsInt, SupportsIndex, str, bytes] = ...
+    ) -> None: ...
 
 class int64(signedinteger):
-    def __init__(self, __value: SupportsInt = ...) -> None: ...
+    def __init__(
+        self, __value: Union[SupportsInt, SupportsIndex, str, bytes] = ...
+    ) -> None: ...
 
 class timedelta64(signedinteger):
-    def __init__(self, __value: Any = ..., __format: str = ...) -> None: ...
+    def __init__(
+        self,
+        __value: Union[None, int, str, bytes, dt.timedelta, timedelta64] = ...,
+        __format: Union[str, bytes] = ...,
```

Comment:
```suggestion
        __format: Union[str, bytes, Tuple[Union[str, bytes], int]] = ...,
```
Since `np.timedelta64(0, ('ms', 3))` is legal. Perhaps an alias for `Union[str, bytes]` would be useful here and elsewhere.

Reply:
Fixed and added an alias: `_CharLike = Union[str, bytes]` (see https://github.com/numpy/numpy/pull/17214/commits/6a3bdeb81e3cc7530d10269f09c414c874f44866).